components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -829,6 +833,28 @@ protected void findPreviousAuthenticatedSession(HttpServletRequest request, Auth
         }
         // Get service provider chain
         SequenceConfig effectiveSequence = getSequenceConfig(context, request.getParameterMap());
+        ServiceProviderProperty[] spProperties = effectiveSequence.getApplicationConfig().getServiceProvider()
+                .getSpProperties();
+        boolean showOrganizationAuthenticator = true;
+        for (ServiceProviderProperty property : spProperties) {
+            if (IS_APP_SHARED.equals(property.getName()) && !Boolean.parseBoolean(property.getValue())) {
+                showOrganizationAuthenticator = false;
+                break;
+            }
+        }
+        if (!showOrganizationAuthenticator) {
+            Map<Integer, StepConfig> stepMap = effectiveSequence.getStepMap();
+            if (MapUtils.isNotEmpty(stepMap)) {
+                StepConfig stepConfig = stepMap.get(1);
+                if (stepConfig != null) {
+                    List<AuthenticatorConfig> authenticatorList = stepConfig.getAuthenticatorList();
+                    authenticatorList = authenticatorList.stream()
+                            .filter(authenticatorConfig -> !authenticatorConfig.getName()
+                                    .equals(ORGANIZATION_AUTHENTICATOR)).collect(Collectors.toList());
+                    stepConfig.setAuthenticatorList(authenticatorList);
+                }
+            }
+        }
```

Comment:
What if we encapsulate this logic to a private method. We may only need to pass the `effectiveSequence`.
Then instead of lot of if checks, we could have returned out from that method.